src/main/scala/cognite/spark/v1/RawTableRelation.scala

```diff
@@ -127,19 +161,27 @@ class RawTableRelation(
         rowConverter(item)
       },
       (r: RawRow) => r.key,
-      getStreams(filter, partitionCursors),
+      streams,
       deduplicateRows = true // if false we might end up with 429 when trying to update assets with multiple same request
     )
   }
 
   override def buildScan(): RDD[Row] = buildScan(schema.fieldNames, Array.empty)
 
+  // scalastyle:off cyclomatic.complexity
   override def buildScan(requiredColumns: Array[String], filters: Array[Filter]): RDD[Row] = {
     val (minLastUpdatedTime, maxLastUpdatedTime) = filtersToTimestampLimits(filters, "lastUpdatedTime")
     val filteredRequiredColumns = getJsonColumnNames(requiredColumns)
     val filteredSchemaFields = getJsonColumnNames(schema.fieldNames)
     val lengthOfRequiredColumnsAsString = requiredColumns.mkString(",").length
 
+    val keysFilter =
+      filters.flatMap {
+        case EqualTo("key", value) => Array(value.toString())
```

Comment:
I think we should handle at least `Or` as well, and `EqualNullSafe`. Check `SequenceRowsRelation.getSeqFilter`. Possibly `And` as well, where we know that there will be zero rows matching "key = 'foo' AND key = 'bar'".

We should also have tests that cover those different cases.